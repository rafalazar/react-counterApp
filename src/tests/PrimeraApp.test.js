import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';
import React from 'react';

describe('Pruebas en <PrimeraApp />', () => {

    // test('Debe mostrar el mensaje: hola soy un saludo :v', () => {

    //     const saludo = 'Hola soy un saludo :v';

    //     const {getByText} = render( <PrimeraApp saludo={saludo} /> );

    //     expect(getByText(saludo)).toBeInTheDocument();
        
    // });

    test('Debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola soy un saludo :v';

        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola soy un saludo :v';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={ subtitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
        
    });
    
    
});
