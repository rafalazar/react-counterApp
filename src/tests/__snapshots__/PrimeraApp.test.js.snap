// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<div>
  <h1>
    Hola soy un saludo :v
  </h1>
  <p>
    Soy un subtitulo
  </p>
</div>
`;
